sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2357,9 +2357,14 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
         Literal(calendarInterval.months, YearMonthIntervalType)
       } else {
         assert(calendarInterval.months == 0)
+        val strToFieldIndex = DayTimeIntervalType.dayTimeFields.map(i =>
```

Comment:
I have seen this already in other PR. Does it make sense to put it to a common place `case object DayTimeIntervalType`?

Reply:
Yeah, it seems a common logic. After this and #32893 are merged, I'll open a followup PR.